roles/pgbouncer/config/tasks/main.yml

```diff
@@ -8,6 +8,30 @@
     mode: 0750
   tags: pgbouncer, pgbouncer_conf
 
+- name: Calculate pool_size for defined pgbouncer_pools (if not defined pool_size, then calculate as pgbouncer_default_pool_size)
+  set_fact:
+    defined_pool_size: "{{ defined_pool_size|default(0)|int + item.pool_parameters|regex_replace ('[^0-9]')|default(pgbouncer_default_pool_size, true)|int }}"
+  loop: "{{ pgbouncer_pools }}"
+  tags: pgbouncer_conf, pgbouncer
+
+- name: Calculate totla_pool_size
+  set_fact:
+    total_pool_size: "{{ defined_pool_size|default(0)|int + (postgresql_databases|length|int - pgbouncer_pools|length|int) * pgbouncer_default_pool_size|int }}"
+  tags: pgbouncer_conf, pgbouncer
+
+- name: Check max_connections
+  set_fact:
+    max_connections: "{{ item.value }}"
+  when: item.option == "max_connections"
```

Comment:
if the parameter is not defined, it may be worth specifying the default value of 100

Reply:
@artemsafiyulin if the max_connections parameter is not specified in postgresql_parameters, I understand there will be an error?

Reply:
i fix it